### ISSUE TITLE
Can this plugin run without the WP JS optimization?

### ISSUE THREAD

**Issue by azirer** [o10n-x/wordpress-css-optimization#24]
If the wordpress-javascript-optimization is not activated this plugin throws a js error from the file `/wordpress-css-optimization/core/controllers/client.class.php` line 
`print '<script data-o10n>o10n.f();</script>';`

In addition i get a PHP error only when the js plugin is not activated
`PHP Fatal error:  Uncaught Error: Call to a member function autoload() on null in ..../public_html/wp-content/plugins/wordpress-css-optimization/core/controllers/core.class.php:254
Stack trace:
#0 [internal function]: O10n\Core->autoload('O10n\\HTMLMinify')
#1 [internal function]: spl_autoload_call('O10n\\HTMLMinify')
#2 ..../public_html/wp-content/plugins/wordpress-html-optimization/controllers/html.class.php(787): class_exists('O10n\\HTMLMinify')
#3 ..../public_html/wp-includes/class-wp-hook.php(286): O10n\Html->process_html('<!DOCTYPE html>...')
#4 ..../public_html/wp-includes/plugin.php(203): WP_Hook->apply_filters('\n<!DOCTYPE html...', Array)
#5 ..../public_html/wp-content/plugins/wordpress-css-optimization/core/controllers/output.class.php(97): apply_filters('o10n_html', '\n<!DOCTYPE html...')
#6 [internal function]: O10n\Output->process_buffer('\n<!DOCTYPE html...', 9)
#7 ..../public_html/wp-includes/functions.php(3730): ob_end_flush()
#8 ..../public_html/w in ..../public_html/wp-content/plugins/wordpress-css-optimization/core/controllers/core.class.php on line 254`

**Comment by optimalisatie** [comment]
Thanks a lot for providing feedback!

In regards to the javascript error, could you please let us know what plugins are enabled? 

In regards to the PHP error. That is indeed a bug. The class autoloader is executed for a class that should not be autoloaded. This only happens when the plugin is disabled while other plugins are being used.

The issue will be fixed in the next update.

**Comment by azirer** [comment]
I have enabled Html Optimization, JS optimization and CSS optimization. among other plugins. 
I have created a test page (It's alive site) where all plugins are disabled except those mentioned above. The frontend doesn't throw any js errors. However if I add the JS plugin to the disabled plugins then I get 
`Uncaught ReferenceError: o10n is not defined`

Have you tried to have activated the CSS plugin without activating the JS plugin? Does it work for you? If yes then I need to setup a similar dev environment and see what causes this.
